L2-330: Confirm update delay

lmuntanerDfinity opened the pull request:
# Motivation

Show confirmation screen to set dissolve delay after creating a neuron.

# Changes

* New step in CreateNeuronModal: `ConfirmDissolveDelay`.
* `SetDissolveDelay` does not update delay, it passes delay to the confirmation screen.
* Add method to set the current step on the StateStep service.

# Tests

* Upgrade CreateNeuronModal tests.
* Add test in StateStep service.

lmuntanerDfinity commented:
New Screen:

![Screenshot 2022-03-14 at 08 35 24](https://user-images.githubusercontent.com/98735375/158133960-53d669ac-a05c-44a0-8711-f30592861f89.png)

lmuntanerDfinity commented:
@peterpeterparker @mstrasinskis please review

peterpeterparker commented on frontend/svelte/src/lib/modals/neurons/CreateNeuronModal.svelte:
```diff
@@ -31,13 +35,17 @@
   const goBack = () => {
     stateStep = stateStep.back();
   };
+  const goNext = () => {
```
can you remove `chooseAccount` that does the same action?

peterpeterparker commented on frontend/svelte/src/lib/modals/neurons/CreateNeuronModal.svelte:
```diff
@@ -31,13 +35,17 @@
+  };
   const goToDissolveDelay = ({
     detail,
   }: CustomEvent<{ neuronId: NeuronId }>) => {
```
code goodie: params of `goToDissolveDelay` can be removed since not used anymore

lmuntanerDfinity replied:
Why? I still need the neuronId when I go to the dissolveDelay screen, what do you mean I can remove?

peterpeterparker replied:
Forget this comment, the github review view made it misleading.

lmuntanerDfinity replied:
Actually, I was able to remove it when I moved the delayInSeconds to the parent. Thanks!

This way, when the user moves back, the delays stays the same.

peterpeterparker reviewed (commented):
Is it correct, there is no back button on confirmation modal?

<img width="1041" alt="Capture d’écran 2022-03-14 à 11 31 16" src="https://user-images.githubusercontent.com/16886711/158154814-58c6480e-a9b2-4391-a1ba-07059c73d4a0.png">

lmuntanerDfinity commented:
@peterpeterparker thanks! Added the button and I also moved the delayInSeconds to the parent, this way I could remove the function that you mentioned!

peterpeterparker commented on frontend/svelte/src/lib/modals/neurons/CreateNeuronModal.svelte:
```diff
@@ -31,13 +35,22 @@
+  let delayInSeconds: number = 0;
+  let showBackButton: boolean;
+  $: showBackButton =
+    currentStep === Steps.StakeNeuron ||
```
I thought you said you like map to map variables 😉

just a comment, it is fine as it is.

can also be `[Steps.StakeNeuron, Steps.ConfirmDisseolveDelay].includes(currentStep)` if you add more steps in the future but again, fine as it is.

lmuntanerDfinity replied:
I like it! Good point

peterpeterparker reviewed (approved):
Thanks
